include/rocksdb/utilities/cache_dump_load.h

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <set>
+
+#include "rocksdb/cache.h"
+#include "rocksdb/env.h"
+#include "rocksdb/io_status.h"
+#include "rocksdb/secondary_cache.h"
+#include "rocksdb/table.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class CacheDumpWriter {
+ public:
+  virtual ~CacheDumpWriter() = default;
+
+  virtual IOStatus Write(const Slice& data) = 0;
+  virtual IOStatus Close() = 0;
+  virtual uint64_t GetFileSize() = 0;
+};
+
+class CacheDumpReader {
+ public:
+  virtual ~CacheDumpReader() = default;
+  virtual IOStatus Read(size_t len, std::string* data) = 0;
+  virtual size_t GetOffset() const = 0;
+  virtual IOStatus Close() = 0;
+};
+
+// CacheDumpOptions is used to control DumpCache()
+struct CacheDumpOptions {
+  // The env pointer
+  Env* env;
+  // The prefix set, which filter out the cache keys that does not match
```

Comment:
This seems like an implementation detail that we don't want to expose in the public API. Perhaps we want filtering that is not prefix matching.

Reply:
Or perhaps we want to prefix match against a Bloom filter instead of an explicit set (for example)

Reply:
@anand1976 @pdillinger Maybe, we can provide a public API in CacheDump called SetDumpFilter(std::vector<DB*> db_list). It take a list of DB pointers and generate the filter internally. In this we, we hide the filter details to application and dump the blocks belonging to certain DB we want. What do you think? It will be easier to extend, e.g., at the beginning, we use prefix_set to search, later on, we can easily change to BF or other more advanced implementations.